src/sources/kafka.rs

```diff
@@ -183,11 +186,12 @@ async fn kafka_source(
     while let Some(message) = stream.next().await {
         match message {
             Err(error) => {
-                emit!(&KafkaEventFailed { error });
+                emit!(&KafkaReadError { error });
             }
             Ok(msg) => {
-                emit!(&KafkaEventReceived {
-                    byte_size: msg.payload_len()
+                emit!(&BytesReceived {
+                    byte_size: std::mem::size_of_val(&msg),
```

Comment:
I'm pretty sure this is wrong. This will report the size of the Rust type. What we want here is the count of raw bytes off the wire from Kafka.